### ISSUE TITLE
[autocomplete] Command Palette accessibility: Focus indicator relies on colour alone

### ISSUE THREAD

**Issue by grace-snow** [mui/base-ui#4182]
# Bug report

## Current behavior

As keyboard focus moves through options in the Command Palette dialog, the only visual indication of which item is focused is a minor colour change to some of the text and to the background colour. As a keyboard user I find this very hard to see. 

According to [WCAG 2.2 Success Criterion 1.4.1 Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html) if colour alone is used to indicate meaning visually, the change must have a contrast difference to the previous state of at least 3:1 ratio. The text colour changes from #7F8287 to #3A3C44, which is a contrast change of only 2.9:1. 

(I have been unable to determine the contrast of the background colour change as it is too subtle to be visible on my screen)

## Expected behavior

Keyboard focus should be clearly visible/obvious. In other autocomplete variants, the background of the focused item changes to a very dark colour, and the text becomes light, making it easy to see where I am focused with the keyboard. Alternatively, a visible focus outline would serve the same purpose.

## Reproducible example

Steps to reproduce:
1. Navigate to the [Autocomplete command palette demo](https://base-ui.com/react/components/autocomplete#command-palette)
2. Using a keyboard, open the command palette 
3. Use Arrow keys to move focus between the suggested options
4. Note the subtle colour change to some of the text (and background if you can see it)
5. Use a tool such as the [Color Contrast Analyser (CCA)](https://vispero.com/color-contrast-checker/) to colour pick colours. Note the contrast differences between focused and unfocused options.

## Base UI version

v1.2.0

## Which browser are you using?

Chrome

## Which OS are you using?

Mac OS

## Which assistive tech are you using (if applicable)?

Keyboard

## Additional context

Keyboard focus should be clearly visible in all component demos

**Comment by atomiks** [comment]
> if colour alone is used to indicate meaning visually, the change must have a contrast difference to the previous state of at least 3:1 ratio. The text colour changes from #7F8287 to #3A3C44, which is a contrast change of only 2.9:1.

Do you mean the background color of the highlighted item needs to be 3:1 with the background color of the non-highlighted items?

Chrome handles this with a left blue border, though it uses the `highlightItemOnHover={false}` behavior which leads to duplicate styles when mixing keyboard & pointer usage, which we're trying to avoid by default:

<img width="617" height="128" alt="Chrome Omnibox virtual highlights" src="https://github.com/user-attachments/assets/7d9b29f6-5254-40da-95e4-a0f2f66936f8" />

**Comment by grace-snow** [comment]
It's actually a little more complex than just colour. Ideally, this needs a `:focus-visible` style such as an outline in addition to the background colour change, as this would be an additional visual indicator of the focus state that doesn't rely on only a colour change. 

If relying on a background colour change, the background colour must present a difference in **hue** as well as having a 3:1 contrast against an unfocused item. 

See the [Understanding WCAG 2.2 Success Criterion 1.4.1 Use of Color](https://www.w3.org/WAI/WCAG22/Understanding/use-of-color#:~:text=If%20content%20is,red%20for%20invalid.) documentation:

> If content is conveyed through the use of colors that differ not only in their hue, but that also have a significant difference in lightness, then this counts as an additional visual distinction, as long as the difference in relative luminance between the colors leads to a contrast ratio of 3:1 or greater. For example, a light green and a dark red differ **both** by color (hue) **and** by lightness, so they would pass if the contrast ratio is at least 3:1. Similarly, if content is distinguished by inverting an element's foreground and background colors, this would pass (again, assuming that the foreground and background colors have a sufficient contrast).

> However, if content relies on the user's ability to accurately perceive or differentiate a particular color an additional visual indicator will be required regardless of the contrast ratio between those colors. For example, knowing whether an outline is green for valid or red for invalid.

**Comment by aarongarciah** [comment]
@grace-snow we recently revamped all demos with a new design. The background contrast for the [Autocomplete command palette demo](https://base-ui.com/react/components/autocomplete#command-palette) is now way more noticeable.
